integration-intygstjanst/src/main/java/se/inera/intyg/minaintyg/integration/intygstjanst/CertificateStatusService.java

```diff
@@ -17,23 +17,37 @@ public List<CertificateStatusType> get(List<CertificateRelationDTO> relations,
       CertificateRecipientDTO recipient,
       LocalDateTime issued) {
 
-    final var statuses = getStatusesFromRelations(relations);
-    statuses.add(CertificateStatusFactory.sent(recipient));
-    statuses.add(CertificateStatusFactory.newStatus(issued));
+    final var statuses = getReplacedStatusFromRelations(relations);
 
-    return statuses
-        .stream()
-        .filter(Optional::isPresent)
-        .map(Optional::get)
-        .toList();
+    if (notReplaced(statuses)) {
+      statuses.add(CertificateStatusFactory.sent(recipient));
+      statuses.add(CertificateStatusFactory.newStatus(issued));
+    }
+
+    return getListOfStatuses(statuses);
```

Comment:
You could remove this private method and keep the stream-logic directly in this method.